File: f77/driver/prn_data.f
C
      Subroutine Print_Data
C
C     This subroutine writes print data to standard output.
C
C     Inputs:
C        DtPrint          - Time step between printing data [sec]
C        Number_Of_Prints - Number of communications array locations to print [Integer]
C        Print_Index      - Communications array location of print data [Integer Array]
C        Real_Array       - The communications array for real variables [Real Array]
C        Time             - Simulation time [sec]
C
C     Outputs:
C        LOG_PRINT_DATA   - Event corresponding to printing data to standard output file [Integer]
C
C     Internal variables and constants:
C        I                - Loop index [Integer]
C
C     Include common block definitions
C
      Include 'global.inc'
      Include 'sysvars.inc'
      Include 'print.inc'
C
C     Event designation parameters
C
      Include 'event.prm'
C
C     Exceptions to default type
C
      Integer I
C
C       Write run data to standard output
C
      write(*, fmt="(ES13.5)", advance="no") Time
      Do i=1,Number_Of_Prints
          write(*, fmt="(ES13.5)", advance="no")
     &        Real_Array(Print_Index(I))
      end do
      Write(*,*)
C
C     Schedule next print
C
      Call Schedule(((Nint(Time/DtPrint) + 1)*DtPrint),LOG_PRINT_DATA)
C
      Return
      End

File: compcert/driver/prn_data.c
#include <math.h>
#include "../driver/prn_data.h"
#include "../driver/global.h"
#include "../driver/sysvars.h"
#include "../driver/print.h"
#include "../driver/event.h"
#include "../driver/schedule.h"
void print_data() {
  // This subroutine writes print data to standard output.
  //
  // Inputs:
  //    DtPrint          - Time step between printing data [sec]
  //    Number_Of_Prints - Number of communications array locations to print [Integer]
  //    Print_Index      - Communications array location of print data [Integer Array]
  //    Real_Array       - The communications array for real variables [Real Array]
  //    Time             - Simulation time [sec]
  //
  // Outputs:
  //    LOG_PRINT_DATA   - Event corresponding to printing data to standard output file [Integer]
  // Write run data to standard output
  printf("%.4f",time);
  for (int i = 0; i < number_of_prints; i++)
    printf("\t%.6f", real_array[print_index[i]].r);
  printf("\n");
  // Schedule next print
  // CALL SCHEDULE(((NINT(time/dtprint)+1)*dtprint),log_print_data)
  schedule((roundf(time/dtprint) + 1.0)*dtprint, LOG_PRINT_DATA);
}
